### PR TITLE
Feature/22 slt user config

### LINKED ISSUE #22: Feature: read SRM model from json
We should separate code from configuration and simplify the definition of SRM logic tree and aslo simplify the way SRM logic trees are defined by the user. This does not change internal classes, it is only a simplification of the config for the user.

- [x] function to read json file equivelent to `nzshm_model.source_logic_tree.slt_config.from_config()
- [x] translate json config to python config (`logic_tree_permuations` and `src_correlations`)
- [x] check that correlations are done correctly (every branch named exists and the primary and secondary fault systems are the same for every pair)

Questions:
should `"fault_system"` be `"branch_set"` instead to be consistent with OQ nrml?

NB: For correlations we no longer specify which group is "dropped". The primary and secondary fault systems are explicitly defined in each correlation on the list.

config.json definition:
```
{
    "title": "a SRM logic tree",
    "version": "SRM_v10.4.0",
    "fault_systems": [
        {
            "name": "Hik",
            "branches": [
                {
                    "tag":  "Hik TL, N16.5, b0.95, C4, s0.42",
                    "weight": 0.147216637218474,
                    "inv_id": "SW52ZXJzaW9uU29sdXRpb25Ocm1sOjExNDQyOA==",
                    "bg_id": "RmlsZToxMzA3MzI="
                },
                {
                    "tag": "Hik TL, N16.5, b0.95, C4, s1",
                    "weight": 0.281154911079988,
                    "inv_id": "SW52ZXJzaW9uU29sdXRpb25Ocm1sOjExNDQxNw==",
                    "bg_id": "RmlsZToxMzA3MzQ="
                }
            ]
        },
        {
            "name": "Puy",
            "branches": [
                {
                    "tag": "Puy 0.7, N4.6, b0.902, C4, s0.28",
                    "weight": 0.21,
                    "inv_id": "SW52ZXJzaW9uU29sdXRpb25Ocm1sOjExODcxNw==",
                    "bg_id": "RmlsZToxMzA3NTM="
                  },
                  {
                    "tag": "Puy 0.7, N4.6, b0.902, C4, s1",
                    "weight": 0.52,
                    "inv_id": "SW52ZXJzaW9uU29sdXRpb25Ocm1sOjExODcyMw==",
                    "bg_id": "RmlsZToxMzA3NTQ="
                  }
            ]
        }
    ],

    "correlations": [
        {
            "primary": {"group": "Hik", "tag": "Hik TL, N16.5, b0.95, C4, s0.42"},
            "secondary": {"group": "Puy", "tag": "Puy 0.7, N4.6, b0.902, C4, s0.28"}
        },
        {
            "primary": {"group": "Hik", "tag": "Hik TL, N16.5, b0.95, C4, s1"},
            "secondary": {"group": "PUY", "tag": "Puy 0.7, N4.6, b0.902, C4, s1"}
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by chrisdicaprio**
Add ability to create new SourceLogicTree and GMCMLogicTree from a hand-built json file. includes examples and documentation.

closes #22

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## [0.9.2] 2024-02-26
+
+### Added
+ - create SourceLogicTree can from hand-built json
```
Comment (chrisbc) [inline CHANGELOG.md:6]: typo ?

File: tests/test_source_lt_from_config.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+
+from nzshm_model.logic_tree import InversionSource, SourceLogicTree
+
+
+def test_slt_from_config():
```
Comment (chrisbc) [inline tests/test_source_lt_from_config.py:6]: @chrisdicaprio suggest some edge case tests here - showing errors thrown for invalid setup ??

File: nzshm_model/logic_tree/source_logic_tree/version2/logic_tree.py [hunk 3]
```diff
@@ -2,12 +2,14 @@
...
 import copy
+import json
 import warnings
 from dataclasses import dataclass, field
+from pathlib import Path
 from typing import Dict, List, Optional, Type, Union
 
 from nzshm_model.logic_tree.correlation import Correlation, LogicTreeCorrelations
-from nzshm_model.logic_tree.logic_tree_base import Branch, BranchSet, FilteredBranch, LogicTree
+from nzshm_model.logic_tree.logic_tree_base import Branch, BranchSet, FilteredBranch, LogicTree, _TLogicTree
```
Comment (chrisbc) [inline nzshm_model/logic_tree/source_logic_tree/version2/logic_tree.py:12]: better to redefine than to `import _TLogicTree` ??

File: nzshm_model/logic_tree/logic_tree_base.py [hunk 4]
```diff
@@ -130,39 +110,70 @@ def combined_branches(self) -> Generator[CompositeBranch, None, None]:
...
+            data = json.load(jsonfile)
+        return cls.from_dict(data)
 
     @classmethod
-    def from_json(cls, json_path: Union[Path, str]) -> 'LogicTree':
+    def from_dict(cls: Type[LogicTreeType], data: Dict) -> LogicTreeType:
         """
-        Create LogicTree object from json file or string
+        Create LogicTree object from dict. This function does not de-serialze direct dict representation of a LogicTree,
+        instead it works on a simlified representation of correaltations meant for user-built logic trees.
```
Comment (chrisbc) [inline nzshm_model/logic_tree/logic_tree_base.py:149]: @chrisdicaprio  typos  `deserialize, simplified` but actually... is this info correct?? 

I thought the intent was that it DOES handle the to_dict() -> from_dict() round trip??

maybe it does (below)
